<a href="https://colab.research.google.com/github/Khadija-Azam05/ML-Projects/blob/main/Copy_of_w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Khadija-Azam05/ML-Projects/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## Lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because it focuses on a real business decision, identifying which content pages should be reviewed first. The starter dataset contains search, traffic, engagement, and content-related signals that can help prioritize pages for review. I am interested in building a machine learning solution that supports human decision-making by ranking pages based on evidence rather than relying only on fixed rules.

No code is needed for this section.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

My research question is:

**Which content pages should be reviewed first based on search and engagement data?**

This project aims to help the SEO or content team decide which pages should be reviewed first. The final output will be a ranked list of pages, allowing the team to decide whether a page should be refreshed, improved, monitored, or left as it is.

If the recommendations are wrong, the team could spend time reviewing pages that don't really need attention, while important pages might be missed. Machine learning can help because it can combine different signals, such as impressions, clicks, CTR, content age, and engagement, to find patterns that are difficult to capture with simple rules. The goal is to support people in making better decisions, not to replace their judgment.

No code is needed for this section.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

## Quick Look at the Data

To support my choice of lane, I explored the starter dataset and looked at a few basic statistics. These numbers will help me understand the data and confirm that it contains enough information to study content refresh opportunities.


In [ ]:
!git clone https://github.com/Khadija-Azam05/ML-Projects.git

Cloning into 'ML-Projects'...
remote: Enumerating objects: 125, done.
remote: Counting objects: 100% (125/125), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 125 (delta 38), reused 94 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (125/125), 1.85 MiB | 9.70 MiB/s, done.
Resolving deltas: 100% (38/38), done.


In [ ]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average sessions (90d):", round(df["sessions_90d"].mean(), 2))
print("Pages with declining trend:", (df["trend_direction"] == "down").sum())

Number of rows: 30000
Number of columns: 44
Average impressions (90d): 5200.37
Average sessions (90d): 37.07
Pages with declining trend: 16262


I explored the starter dataset before choosing my lane. It contains **30,000 pages** and **44 features**. The average impressions over 90 days are **5,200.37**, the average sessions are **37.07**, and **16,262 pages** show a declining trend. These numbers suggest the dataset has enough information to explore content refresh opportunities.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will identify pages that may be good candidates for review based on observed search and engagement data. It is meant to support decision-making, not replace it. I cannot claim that refreshing a page will guarantee better rankings or more traffic, or that I can predict Google's ranking algorithm.

No code is needed for this section.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.